libr/anal/esil.c

```diff
@@ -3390,8 +3387,6 @@ R_API bool r_anal_esil_setup(RAnalEsil *esil, RAnal *anal, int romem, int stats,
 	r_anal_esil_stats (esil, stats);
 	r_anal_esil_setup_ops (esil);
 
-	if (anal->cur && anal->cur->esil_init && anal->cur->esil_fini) {
-		return anal->cur->esil_init (esil);
-	}
-	return true;
+	return (anal->cur && anal->cur->esil_init)
```

Comment:
It should be the same as:
```suggestion
	return !anal->cur || !anal->cur->esil_init || anal->cur->esil_init (esil);
```

Reply:
i think its more readable the way it is now